src/main/kotlin/com/msd/robot/application/RobotApplicationService.kt

```diff
@@ -0,0 +1,53 @@
+package com.msd.robot.application
+
+import com.msd.application.ClientException
+import com.msd.application.GameMapService
+import com.msd.command.MovementCommand
+import com.msd.robot.domain.NotEnoughEnergyException
+import com.msd.robot.domain.PlanetBlockedException
+import com.msd.robot.domain.RobotRepository
+import org.springframework.stereotype.Service
+
+@Service
+class RobotApplicationService(
+    val robotRepo: RobotRepository,
+    val gameMapService: GameMapService
+) {
+
+    fun move(moveCommand: MovementCommand) {
+        val robotId = moveCommand.robotId
+        val playerId = moveCommand.playerUUID
+
+        val robotOptional = robotRepo.findById(robotId)
+        if (!robotOptional.isPresent) {
```

Comment:
There is probably a Kotlin way to do this rather than using optionals.

Reply:
It would mean I'd have to add an extra method in the repository to return a nullable instead of an optional, and then use an elvis operator in combination with the run { } method. I can do that, but I'm not sure if that's worth it

Reply:
not an elegant one

Reply:
Probably a bettet one then Java Optionals. Take a look at [this](https://stackoverflow.com/a/53672633).

Reply:
I'd still prefer it if you did it the kotlin way.